internal/layer/metadata_restorer.go

```diff
@@ -12,23 +12,23 @@ import (
 	"github.com/buildpacks/lifecycle/platform"
 )
 
-//go:generate mockgen -package testmock -destination testmock/metadata_restorer.go github.com/buildpacks/lifecycle/internal/layer MetadataRestorer
+//go:generate mockgen -package testmock -destination ../../testmock/metadata_restorer.go github.com/buildpacks/lifecycle/internal/layer MetadataRestorer
 type MetadataRestorer interface {
 	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata, layerSHAStore SHAStore) error
 }
 
 func NewMetadataRestorer(logger Logger, layersDir string, skipLayers bool) MetadataRestorer {
 	return &DefaultMetadataRestorer{
-		logger:     logger,
-		layersDir:  layersDir,
-		skipLayers: skipLayers,
+		Logger:     logger,
+		LayersDir:  layersDir,
+		SkipLayers: skipLayers,
```

Comment:
These could maybe be made private

Reply:
I made them public so they could be tested here: https://github.com/buildpacks/lifecycle/blob/8b9e787a0028978da28976fc6d63a4de82bc9dc0/analyzer_test.go#L278-L280